jetty-webapp/src/test/java/org/eclipse/jetty/webapp/WebAppContextTest.java

```diff
@@ -97,6 +103,53 @@ public void testSessionListeners ()
         assertEquals(1, listeners.size());
     }
     
+    @Test
+    public void testMaxFormParamsViaContextParams() throws Exception
+    {
+        // Initialize a WebAppContext
+        Path testWebappDir = MavenTestingUtils.getProjectDirPath("src/test/webapp");
+        Resource webapp = new PathResource(testWebappDir);
+        WebAppContext context = new WebAppContext();
+        context.setBaseResource(webapp);
+        context.setContextPath("/test");
+
+        // Load a web.xml setting org.eclipse.jetty.server.Request.maxFormContentSize to 100
+        File webMaxForm = MavenTestingUtils.getTestResourceFile("web_maxform.xml");
+        context.getMetaData().setWebXml(Resource.newResource(webMaxForm));
+
+        Server server = new Server();
+        context.setServer(server);
+        server.setHandler(context);
+
+        context.setSessionHandler(new SessionHandler()
```

Comment:
Having this as a SessionHandler seems odd.
Perhaps a `context.addServlet()` with a servlet would be a better fit (testing wise)